### ISSUE TITLE
IntelliSense experiences "Program terminated with signal SIGKILL, Killed."

### ISSUE THREAD

**Issue by EmbeddedBacon** [microsoft/vscode-cpptools#11003]
### Environment

- OS and Version:  Windows 10 Enterprise
- VS Code Version:  1.78.2
- C/C++ Extension Version:  v1.15.4
- If using SSH remote, specify OS of remote machine: Fedora release 34 (Thirty Four)

### Bug Summary and Steps to Reproduce

Bug Summary: The language server that is running on the remote machine is restarting many times during coding sessions.   The log in VS Code shows "[Error - 12:59:30 PM] The language server crashed. Restarting...".    Attaching the debugger yields the a log entry of "Program terminated with signal SIGKILL, Killed."

Based upon comments supplied and followed from https://github.com/microsoft/vscode-cpptools/issues/10636, the following values were changed and currently set to:  C_Cpp.intelliSenseMaxMemory = 65536, C_Cpp.intelliSenseCacheSize = 5120, and C_Cpp.intelliSenseMemoryLimit = 16384

At one time C_Cpp.intelliSenseCacheSize was set to zero, but that didn't seem to help either.

Steps to reproduce:
1. Go to remote to system and open the project folder.

### Configuration and Logs

```shell
----Start of cpptools tab log------
[Error - 12:59:30 PM] Sending request cpptools/getSemanticTokens failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getSemanticTokens failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getDocumentSymbols failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getSemanticTokens failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getDocumentSymbols failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getDocumentSymbols failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/definition failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getDocumentSymbols failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/documentHighlight failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getCodeActions failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getDocumentSymbols failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/hover failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/documentHighlight failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getCodeActions failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/hover failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getCodeActions failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getDocumentSymbols failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/documentHighlight failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getCodeActions failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getCodeActions failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getDocumentSymbols failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/hover failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/hover failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/documentHighlight failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getCodeActions failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getCodeActions failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getDocumentSymbols failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/documentHighlight failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getCodeActions failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getCodeActions failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getDocumentSymbols failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/hover failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/hover failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getDocumentSymbols failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getDocumentSymbols failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/hover failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/hover failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getCodeActions failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/hover failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/hover failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/definition failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/hover failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getDocumentSymbols failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getCodeActions failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getDocumentSymbols failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/documentHighlight failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/documentHighlight failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getCodeActions failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getCodeActions failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getCodeActions failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getCodeActions failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getDocumentSymbols failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getDocumentSymbols failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getDocumentSymbols failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/hover failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/hover failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/hover failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/hover failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/hover failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/hover failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/hover failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/hover failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/hover failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/documentHighlight failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getCodeActions failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/documentHighlight failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getCodeActions failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/hover failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/hover failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/hover failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/documentHighlight failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getCodeActions failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/hover failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/documentHighlight failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getCodeActions failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/hover failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/hover failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/hover failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/hover failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/documentHighlight failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getCodeActions failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getCodeActions failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/hover failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/hover failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/hover failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/hover failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/documentHighlight failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getCodeActions failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/hover failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/hover failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getCodeActions failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/hover failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/documentHighlight failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getCodeActions failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/hover failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/hover failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/hover failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/hover failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getCodeActions failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/definition failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/hover failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/hover failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/hover failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getCodeActions failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/hover failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/documentHighlight failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getCodeActions failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/hover failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getCodeActions failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getCodeActions failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getCodeActions failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getCodeActions failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request textDocument/hover failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] Sending request cpptools/getInlayHints failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097
[Error - 12:59:30 PM] The language server crashed. Restarting...
----End of cpptools tab log------



----Start Log from attaching debugger to bin/cpptools------
=thread-group-added,id="i1"
GNU gdb (GDB) Fedora 10.1-14.fc34
Copyright (C) 2020 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64-redhat-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<https://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word".
=cmd-param-changed,param="history save",value="on"
=cmd-param-changed,param="print pretty",value="on"
=cmd-param-changed,param="print array-indexes",value="on"
=cmd-param-changed,param="python print-stack",value="full"
=cmd-param-changed,param="pagination",value="off"
=cmd-param-changed,param="auto-load safe-path",value="/"
Warning: Debuggee TargetArchitecture not detected, assuming x86_64.
[New LWP 169156]
e[He[Je[3Je[36m───e[0m e[1;33mOutput/messagese[0m e[36m────────────────────────────────────────────────────────────e[0m
[New LWP 169157]
e[He[Je[3Je[36m───e[0m e[1;33mOutput/messagese[0m e[36m────────────────────────────────────────────────────────────e[0m
[New LWP 169158]
e[He[Je[3Je[36m───e[0m e[1;33mOutput/messagese[0m e[36m────────────────────────────────────────────────────────────e[0m
[New LWP 169159]
e[He[Je[3Je[36m───e[0m e[1;33mOutput/messagese[0m e[36m────────────────────────────────────────────────────────────e[0m
[New LWP 169160]
e[He[Je[3Je[36m───e[0m e[1;33mOutput/messagese[0m e[36m────────────────────────────────────────────────────────────e[0m
[New LWP 169161]
e[He[Je[3Je[36m───e[0m e[1;33mOutput/messagese[0m e[36m────────────────────────────────────────────────────────────e[0m
[New LWP 169162]
e[He[Je[3Je[36m───e[0m e[1;33mOutput/messagese[0m e[36m────────────────────────────────────────────────────────────e[0m
[New LWP 169163]
e[He[Je[3Je[36m───e[0m e[1;33mOutput/messagese[0m e[36m────────────────────────────────────────────────────────────e[0m
[New LWP 169164]
e[He[Je[3Je[36m───e[0m e[1;33mOutput/messagese[0m e[36m────────────────────────────────────────────────────────────e[0m
[New LWP 169165]
e[He[Je[3Je[36m───e[0m e[1;33mOutput/messagese[0m e[36m────────────────────────────────────────────────────────────e[0m
[New LWP 169166]
e[He[Je[3Je[36m───e[0m e[1;33mOutput/messagese[0m e[36m────────────────────────────────────────────────────────────e[0m
[New LWP 169167]
e[He[Je[3Je[36m───e[0m e[1;33mOutput/messagese[0m e[36m────────────────────────────────────────────────────────────e[0m
[New LWP 169168]
e[He[Je[3Je[36m───e[0m e[1;33mOutput/messagese[0m e[36m────────────────────────────────────────────────────────────e[0m
[New LWP 169169]
e[He[Je[3Je[36m───e[0m e[1;33mOutput/messagese[0m e[36m────────────────────────────────────────────────────────────e[0m
[New LWP 169170]
e[He[Je[3Je[36m───e[0m e[1;33mOutput/messagese[0m e[36m────────────────────────────────────────────────────────────e[0m
[New LWP 169171]
e[He[Je[3Je[36m───e[0m e[1;33mOutput/messagese[0m e[36m────────────────────────────────────────────────────────────e[0m
[New LWP 169172]
e[He[Je[3Je[36m───e[0m e[1;33mOutput/messagese[0m e[36m────────────────────────────────────────────────────────────e[0m
[New LWP 169173]
e[He[Je[3Je[36m───e[0m e[1;33mOutput/messagese[0m e[36m────────────────────────────────────────────────────────────e[0m
[New LWP 169174]
e[He[Je[3Je[36m───e[0m e[1;33mOutput/messagese[0m e[36m────────────────────────────────────────────────────────────e[0m
[New LWP 169175]
e[He[Je[3Je[36m───e[0m e[1;33mOutput/messagese[0m e[36m────────────────────────────────────────────────────────────e[0m
[New LWP 169323]
e[He[Je[3Je[36m───e[0m e[1;33mOutput/messagese[0m e[36m────────────────────────────────────────────────────────────e[0m
0x00007f4c7da0d31c in read () from /usr/lib64/libc.so.6
e[36m───e[0m e[1;33mAssemblye[0m e[36m───────────────────────────────────────────────────────────────────e[0m
 e[1;32m0x00007f4c7da0d31ce[0me[1;32m  e[0me[1;32me[0me[1;32m?e[0me[1;32m cmp    $0xfffffffffffff000,%raxe[0m
 e[90m0x00007f4c7da0d322e[0m  e[90m?e[0m ja     0x7f4c7da0d358 <read+136>
 e[90m0x00007f4c7da0d324e[0m  e[90m?e[0m mov    %r8d,%edi
 e[90m0x00007f4c7da0d327e[0m  e[90m?e[0m mov    %rax,0x8(%rsp)
 e[90m0x00007f4c7da0d32ce[0m  e[90m?e[0m call   0x7f4c7d9a1d80 <__libc_disable_asynccancel>
 e[90m0x00007f4c7da0d331e[0m  e[90m?e[0m mov    0x8(%rsp),%rax
 e[90m0x00007f4c7da0d336e[0m  e[90m?e[0m add    $0x28,%rsp
 e[90m0x00007f4c7da0d33ae[0m  e[90m?e[0m ret    
 e[90m0x00007f4c7da0d33be[0m  e[90m?e[0m nopl   0x0(%rax,%rax,1)
 e[90m0x00007f4c7da0d340e[0m  e[90m?e[0m mov    0xd1b09(%rip),%rdx        # 0x7f4c7dadee50
e[36m───e[0m e[33mBreakpointse[0m e[36m────────────────────────────────────────────────────────────────e[0m
e[36m───e[0m e[33mExpressionse[0m e[36m────────────────────────────────────────────────────────────────e[0m
e[36m───e[0m e[33mHistorye[0m e[36m────────────────────────────────────────────────────────────────────e[0m
e[36m───e[0m e[33mMemorye[0m e[36m─────────────────────────────────────────────────────────────────────e[0m
e[36m───e[0m e[1;33mRegisterse[0m e[36m──────────────────────────────────────────────────────────────────e[0m
   e[90mraxe[0m e[1;32m0xfffffffffffffe00e[0m   e[90mrbxe[0m e[1;32m0x00007f4c7dadf800e[0m      e[90mrcxe[0m e[1;32m0x00007f4c7da0d31ce[0m
   e[90mrdxe[0m e[1;32m0x0000000000001000e[0m   e[90mrsie[0m e[1;32m0x0000000002b44b30e[0m      e[90mrdie[0m e[1;32m0x0000000000000000e[0m
   e[90mrbpe[0m e[1;32m0x00007f4c7dae1300e[0m   e[90mrspe[0m e[1;32m0x00007fff1be4d870e[0m       e[90mr8e[0m e[1;32m0x0000000000000000e[0m
    e[90mr9e[0m e[1;32m0x000000000168ed78e[0m   e[90mr10e[0m e[1;32m0x0000000000000000e[0m      e[90mr11e[0m e[1;32m0x0000000000000246e[0m
   e[90mr12e[0m e[1;32m0x00007fff1be4d9b8e[0m   e[90mr13e[0m e[1;32m0x0000000000000d68e[0m      e[90mr14e[0m e[1;32m0x00007f4c7dae0700e[0m
   e[90mr15e[0m e[1;32m0x0000000000000d68e[0m   e[90mripe[0m e[1;32m0x00007f4c7da0d31ce[0m   e[90meflagse[0m e[1;32m[ PF ZF IF ]e[0m      
    e[90mcse[0m e[1;32m0x00000033e[0m            e[90msse[0m e[1;32m0x0000002be[0m               e[90mdse[0m e[1;32m0x00000000e[0m        
    e[90mese[0m e[1;32m0x00000000e[0m            e[90mfse[0m e[1;32m0x00000000e[0m               e[90mgse[0m e[1;32m0x00000000e[0m        
e[36m───e[0m e[33mSourcee[0m e[36m─────────────────────────────────────────────────────────────────────e[0m
e[36m───e[0m e[1;33mStacke[0m e[36m──────────────────────────────────────────────────────────────────────e[0m
[e[1;32m0e[0m] from e[1;32m0x00007f4c7da0d31ce[0m in e[1;32mreade[0m
[e[32m1e[0m] from e[32m0x00007f4c7d99e9d4e[0m in e[32m_IO_file_underflowe[0m
[e[32m2e[0m] from e[32m0x00007f4c7d99fc26e[0m in e[32m_IO_default_uflowe[0m
[e[32m3e[0m] from e[32m0x00007f4c7d99a268e[0m in e[32mgetce[0m
[e[32m4e[0m] from e[32m0x0000000000ffdf2de[0m in e[32m__gnu_cxx::stdio_sync_filebuf<char, std::char_traits<char> >::underflow()e[0m
[e[32m5e[0m] from e[32m0x0000000000fc10aae[0m in e[32mstd::basic_istream<char, std::char_traits<char> >& std::getline<char, std::char_traits<char>, std::allocator<char> >(std::basic_istream<char, std::char_traits<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, char)e[0m
[e[32m6e[0m] from e[32m0x00000000005ecf79e[0m in e[32mlsp_manager::read_message(lsp_manager::client_message&) conste[0m
[e[32m7e[0m] from e[32m0x00000000005ed7c5e[0m in e[32mlsp_manager::process_messages(std::function<void ()> const&)e[0m
[e[32m8e[0m] from e[32m0x00000000004f5c29e[0m in e[32mcpptools_context::run()e[0m
[e[32m9e[0m] from e[32m0x00000000005f28c9e[0m in e[32mmaine[0m
e[36m───e[0m e[1;33mThreadse[0m e[36m────────────────────────────────────────────────────────────────────e[0m
[e[32m22e[0m] id e[32m169323e[0m name e[32mcpptoolse[0m from e[32m0x00007f4c7db00a8ae[0m in e[32m__futex_abstimed_wait_common64e[0m
[e[32m21e[0m] id e[32m169175e[0m name e[32mcpptoolse[0m from e[32m0x00007f4c7db00a8ae[0m in e[32m__futex_abstimed_wait_common64e[0m
[e[32m20e[0m] id e[32m169174e[0m name e[32mcpptoolse[0m from e[32m0x00007f4c7da1c69ee[0m in e[32mepoll_waite[0m
[e[32m19e[0m] id e[32m169173e[0m name e[32mcpptoolse[0m from e[32m0x00007f4c7db00a8ae[0m in e[32m__futex_abstimed_wait_common64e[0m
[e[32m18e[0m] id e[32m169172e[0m name e[32mcpptoolse[0m from e[32m0x00007f4c7db00a8ae[0m in e[32m__futex_abstimed_wait_common64e[0m
[e[32m17e[0m] id e[32m169171e[0m name e[32mcpptoolse[0m from e[32m0x00007f4c7db00a8ae[0m in e[32m__futex_abstimed_wait_common64e[0m
[e[32m16e[0m] id e[32m169170e[0m name e[32mcpptoolse[0m from e[32m0x00007f4c7db00a8ae[0m in e[32m__futex_abstimed_wait_common64e[0m
[e[32m15e[0m] id e[32m169169e[0m name e[32mcpptoolse[0m from e[32m0x00007f4c7db00a8ae[0m in e[32m__futex_abstimed_wait_common64e[0m
[e[32m14e[0m] id e[32m169168e[0m name e[32mcpptoolse[0m from e[32m0x0000000000710144e[0m in e[32mcode_store::CEncParseNode::CEnum::MoveNext()e[0m
[e[32m13e[0m] id e[32m169167e[0m name e[32mcpptoolse[0m from e[32m0x00007f4c7db00a8ae[0m in e[32m__futex_abstimed_wait_common64e[0m
[e[32m12e[0m] id e[32m169166e[0m name e[32mcpptoolse[0m from e[32m0x00007f4c7db00a8ae[0m in e[32m__futex_abstimed_wait_common64e[0m
[e[32m11e[0m] id e[32m169165e[0m name e[32mcpptoolse[0m from e[32m0x00007f4c7db00a8ae[0m in e[32m__futex_abstimed_wait_common64e[0m
[e[32m10e[0m] id e[32m169164e[0m name e[32mcpptoolse[0m from e[32m0x00007f4c7dafd7b0e[0m in e[32m__lll_lock_waite[0m
[e[32m9e[0m] id e[32m169163e[0m name e[32mcpptoolse[0m from e[32m0x00007f4c7db00a8ae[0m in e[32m__futex_abstimed_wait_common64e[0m
[e[32m8e[0m] id e[32m169162e[0m name e[32mcpptoolse[0m from e[32m0x00007f4c7db00a8ae[0m in e[32m__futex_abstimed_wait_common64e[0m
[e[32m7e[0m] id e[32m169161e[0m name e[32mcpptoolse[0m from e[32m0x00007f4c7db00a8ae[0m in e[32m__futex_abstimed_wait_common64e[0m
[e[32m6e[0m] id e[32m169160e[0m name e[32mcpptoolse[0m from e[32m0x00007f4c7dafd7b0e[0m in e[32m__lll_lock_waite[0m
[e[32m5e[0m] id e[32m169159e[0m name e[32mcpptoolse[0m from e[32m0x00007f4c7db00a8ae[0m in e[32m__futex_abstimed_wait_common64e[0m
[e[32m4e[0m] id e[32m169158e[0m name e[32mcpptoolse[0m from e[32m0x00007f4c7db00a8ae[0m in e[32m__futex_abstimed_wait_common64e[0m
[e[32m3e[0m] id e[32m169157e[0m name e[32mcpptoolse[0m from e[32m0x00007f4c7dafd7b0e[0m in e[32m__lll_lock_waite[0m
[e[32m2e[0m] id e[32m169156e[0m name e[32mcpptoolse[0m from e[32m0x00007f4c7da1c69ee[0m in e[32mepoll_waite[0m
[e[1;32m1e[0m] id e[1;32m169155e[0m name e[1;32mcpptoolse[0m from e[1;32m0x00007f4c7da0d31ce[0m in e[1;32mreade[0m
e[36m───e[0m e[33mVariablese[0m e[36m──────────────────────────────────────────────────────────────────e[0m
e[36m────────────────────────────────────────────────────────────────────────────────e[0m
Loaded '/usr/lib64/libm.so.6'. Symbols loaded.
Loaded '/usr/lib64/libpthread.so.0'. Symbols loaded.
Loaded '/usr/lib64/libc.so.6'. Symbols loaded.
Loaded '/usr/lib64/libdl.so.2'. Symbols loaded.
Loaded '/usr/lib64/librt.so.1'. Symbols loaded.
Loaded '/usr/lib64/libgcc_s.so.1'. Symbols loaded.
Loaded '/lib64/ld-linux-x86-64.so.2'. Symbols loaded.
e[He[Je[3Je[36m───e[0m e[1;33mOutput/messagese[0m e[36m────────────────────────────────────────────────────────────e[0m
[LWP 169323 exited]
[LWP 169175 exited]
[LWP 169174 exited]
[LWP 169173 exited]
[LWP 169172 exited]
[LWP 169171 exited]
[LWP 169170 exited]
[LWP 169169 exited]
[LWP 169168 exited]
[LWP 169167 exited]
[LWP 169166 exited]
[LWP 169165 exited]
[LWP 169164 exited]
[LWP 169163 exited]
[LWP 169162 exited]
[LWP 169161 exited]
[LWP 169160 exited]
[LWP 169159 exited]
[LWP 169158 exited]
[LWP 169156 exited]
[LWP 169155 exited]

Program terminated with signal SIGKILL, Killed.
The program no longer exists.
Debugger has disconnected from the program '/users/mrhicks/.vscode-server/extensions/ms-vscode.cpptools-1.15.4/bin/cpptools'.
----End Log from attaching debugger to bin/cpptools------
```

**Comment by karan-k-deepr** [comment]
I am also getting a similar error while working with `Dev Container`. My system freezes when this happens, I have checked if there is any memory or cpu which is causing the freeze. But it seems the prolonged running of `c++ intellisense` causes this issue.

**Comment by sean-mcmanus** [comment]
@EmbeddedBacon I've reproed this issue several times with 1.16.0 (i.e. probably the same cause as 1.15.4). I'm not sure what triggers/causes it yet though. It only seems to repro in certain situations.

@karan-k-deepr Your issue sounds different. There is no freeze involved, just the cpptools process is exiting when it shouldn't.

**Comment by EmbeddedBacon** [comment]
Mr. @sean-mcmanus , I have often wonder if the sigkill is related to the project size.  The entire project size is 15 GB, including the .git directory which is 4.4 GB.   The project also includes a number of project header files for specific hardware controllers.  The total for these project header files for the "specific hardware controllers" totals 4.5 GB.   

When looking at the little cylinder icon at the bottom of VS Code that shows the current progress of "Parsing working place files", it will generally get stuck around 6-9% from what I can tell.  If I look in the output window for C/C++ tools it always seems stuck "tag parsing file:" while parsing one of the header files in the "specific hardware controllers" directory.

In https://github.com/microsoft/vscode-cpptools/issues/10636 you have already helped a lot trying to diagnose this issue.  We even capture CPU and memory usage which didn't show high activity or large amounts of RAM usage.

Question, I have looked through the documentation and Stack Overflow, but is there a way to exclude directories from being parsed?

If there is any test or data captures you would like me to perform I will be more than willing to perform them.  Thanks, sir.

**Comment by sean-mcmanus** [comment]
The issue I seemed to be hitting was only with 1.16.0 and caused by our thread pool not increasing in size. If you're reproing the bug with 1.15.4, that might be a different root cause bug. After that is fixed, I'll have to see if I still repro the same issue or not.

**Comment by EmbeddedBacon** [comment]
@sean-mcmanus , is there a potential fix on the pre-release branch that I can try?  yes, I have been seeing this issue for many release now, not just 1.15.4.  Currently, I have VSCode configured to use the pre-release branch.

**Comment by sean-mcmanus** [comment]
@EmbeddedBacon Good news -- we believed we figured out the cause of the issue affecting 1.15.4 (and other older/newer versions). It should be available once 1.16.1 is released (we're still working on other changes for it).

**Comment by michelleangela** [comment]
Fixed in pre-release 1.16.1 https://github.com/microsoft/vscode-cpptools/releases/tag/v1.16.1.

**Comment by EmbeddedBacon** [comment]
Been testing pre-release 1.16.1 for the past hour.  Here are the notes so far

1) I don't see the "Program terminated with signal SIGKILL, killed" message so far

2) I had to clear the cache in ~/.cache/vscode-cpptools/ipch as the tools got stuck and the status never got past "IntelliSense: Updating" and the "Parsing Workspace" was paused.   Also, "Outline" pane would say "Loading document symbols for 'filename'", but after several minutes the message of "no symbols found"

3) After clearing the cache the "Outline" pane did populate with the symbols, some what.  For some files the symbols did get loaded, but if I switch to a different file the status of "Loading document symbols for 'filename'" even after waiting some time (3-4 minutes).  

4) In a c source file if I try hovering over a symbol it would show "loading", but never resolve.  In the "output" pane for c/cpp the   "LSP: textDocument/hover" event is seen, but the symbol is never resolve even after not moving the mouse cursor for several minutes.  

5) Pressing F2 / "Go to definition" just results in the magical Cylon eye in the editor tabs, but nothing is found. 

Prior to pre-release 1.16.1 the "Go to definition" would work at times.  Hovering over a symbols some times would result in tooltip showing information for that symbol.  So far I haven't seen either of these with this pre-release.    Also, in the output window for c/cpp there was some activity showing progress of tag parsing head files.   When switching from one source file to another these events are seen

LSP: cpptools/textEditorSelectionChange
LSP: cpptools/textEditorSelectionChange
LSP: cpptools/activeDocumentChange: file:///users/..
LSP: $/cancelRequest (cpptools/getCodeActions, id: 180)
LSP: cpptools/getCodeActions: file:///users/../shared_mem.h (id: 184)
LSP: textDocument/hover: file:///users/../shared_mem.h (id: 185)
LSP: $/cancelRequest (textDocument/hover, id: 185)
LSP: textDocument/hover: file:///users/../shared_mem.h (id: 186)
LSP: $/cancelRequest (textDocument/hover, id: 186)
LSP: textDocument/hover: file:///users/../shared_mem.h (id: 187)
LSP: $/cancelRequest (textDocument/hover, id: 187)
LSP: textDocument/hover: file:///users/../shared_mem.h (id: 188)
LSP: $/cancelRequest (textDocument/hover, id: 188)
LSP: $/cancelRequest (cpptools/getInlayHints, id: 181)
LSP: cpptools/getInlayHints: file:///users/shared_mem.h (id: 189)

**Comment by bobbrow** [comment]
> Question, I have looked through the documentation and Stack Overflow, but is there a way to exclude directories from being parsed?

I don't think this was answered yet.  You can use the `"C_Cpp.files.exclude"` setting to set paths in your workspace that you'd like for us to exclude.

**Comment by Colengms** [comment]
Hi @EmbeddedBacon .

> I had to clear the cache in ~/.cache/vscode-cpptools/ipch as the tools got stuck and the status never got past "IntelliSense: Updating" and the "Parsing Workspace" was paused. Also, "Outline" pane would say "Loading document symbols for 'filename'", but after several minutes the message of "no symbols found"

The IPCH is used by the IntelliSense process.  Clearing it would cause IntelliSense to start up a bit slower per file, as IPCH would need to be regenerated (per file).  Workspace parsing and the Outline are not technically IntelliSense operations.  However, there is a known bug that when an initial IntelliSense update takes a long time, some operations may block synchronously waiting for it to complete.  I suspect that is what you were seeing in the snippet of log you posted.  The logging of receipt and cancellation of the LSP message occurs in a dedicated thread.  But the main thread that processes that work queue may be stalled on a synchronous wait.

For the other anomalous behaviors, could you open a new issue (or multiple), with log output leading up to the repro behavior?  It seems like something may have happened earlier, such as the database being fully reset and needing to repopulate, which might partially explain the other behaviors you're seeing.  Also, you might be able to get some useful information by attaching a debugger to cpptools and grabbing  stacks while it seems to be failing to process certain operations.

**Comment by EmbeddedBacon** [comment]
Mr. @Colengms , thanks for the reply.  I understand about deleting the cache, but I have had to do that from time to time, especially after updating this extension.  Even after 3-4 hours of VS Code being open there still wasn't any change in the status regarding any of the items I listed above.  I say this release isn't doing as well as before.  I can create new reports and I will do that tomorrow.

**Comment by EmbeddedBacon** [comment]
Question: So upon launching VS Code today which the instance is using remote to Fedora system.  I see this in the c/cpp log window.

> Discovering files: 99132 file(s) processed
>   0 file(s) removed from database
> Done discovering files.
> Populating include completion cache.
> Parsing remaining files...
>   tag parsing file: /users/mrhicks/projects/xxx/repo/common/fwassert.h
>   tag parsing file: /opt/synopsys/MetaWare/S-2021.06/MetaWare/arc/inc/arc/arc_intrinsics.h
>   tag parsing file: /users/mrhicks/projects/xxx/repo/common/mem_utils.h
>   tag parsing file: /users/mrhicks/projects/xxx/repo/contract/controller/.../asic/.../..._reg_access.h
> LSP: $/cancelRequest (cpptools/getInlayHints, id: 24)
> LSP: cpptools/textEditorSelectionChange
> LSP: cpptools/textEditorSelectionChange
> LSP: textDocument/didOpen: file:///users/mrhicks/projects/xxx/repo/firmware/src/cs/inc/shared_mem.h

In the past I would see additional log messages with "tag parsing file:" but nothing after this point.  I am not sure if this is okay or not.  Just going off past experience.

Seeing the log entry about "populating include completion cache I checked that folder to see what was there and this is what I see

mrhicks in systemX in .cache/vscode-cpptools/ipch 
🕙 13:35:11 λ  l
Permissions Links Size Blocks User    Group      Date Modified Name
drwxr-xr-x      2    -      - mrhicks nve_collab 20 Jun 12:59  6f725ba0779c9408
drwxr-xr-x      2    -      - mrhicks nve_collab 20 Jun 18:46  7455adc678a731a1


🕙 13:35:40 λ  l
Permissions Links Size Blocks User    Group      Date Modified Name
.rw-r--r--      1 33Mi  68192 mrhicks nve_collab 20 Jun 18:46  fsa_bootFiles.ipch
.rw-r--r--      1    8      8 mrhicks nve_collab 20 Jun 12:59  mmap_address.bin

🕙 13:35:52 λ  l
Permissions Links Size Blocks User    Group      Date Modified Name
.rw-r--r--      1 11Mi  22352 mrhicks nve_collab 20 Jun 12:59  fsa_mgmt.ipch
.rw-r--r--      1    8      8 mrhicks nve_collab 20 Jun 12:59  mmap_address.bin


Only going based upon filenames, was only two source files parsed then?

**Comment by Colengms** [comment]
Hi @EmbeddedBacon .  There are two major components of the C/C++ Extension: The tag parser (or browse database), and IntelliSense.  The tag parser is syntactic/lexical.  It scans for known keywords and constructs, but knows very little about user-defined types.  IntelliSense is semantic.  IntelliSense involves a C++ compiler front-end that actually attempts to compile an open file and provides features based on that understanding of the code.

The IPCH files are related to the semantic IntelliSense pass.  Only after opening a source/header file, and an IntelliSense pass successfully run, is an IPCH generated for that file.  (Actually, headers will resolve to a source file, so there may be fewer than 1:1 per file).

>tag parsing file: /users/mrhicks/projects/xxx/repo/contract/controller/.../asic/.../..._reg_access.h

The fact that tag parsing stalls at this file is worrisome.  This could indicate that some internal error is occurring, such as an infinite loop, while parsing that header.  Could you try running the "Reset IntelliSense Database" command?  (The name of that command is perhaps misleading, as the browse database is not technically related to IntelliSense parsing/compiling).  If the issue persists, you might try excluding that file using `C_Cpp.files.exclude`, to see if that file is directly related to the issue.

**Comment by EmbeddedBacon** [comment]
Instead of adding more comments related to tag parsing here, going to create a new report.  Though, a "Reset IntelliSense Database" was issued, but stalled at a header file.  Granted, these are hardware descriptor header files that are about 213MB in size.

**Comment by EmbeddedBacon** [comment]
@Colengms , how do I determine the status of IntelliSense more so than what is presented here

![image](https://github.com/microsoft/vscode-cpptools/assets/42427056/7cff0767-b586-4efd-96a1-6aa3dd5b6b57)

**Comment by bobbrow** [comment]
If Parsing Workspace is "paused" and IntelliSense is "updating", usually that means your `includePath` setting has a recursive path in it (e.g. `${workspaceFolder}/**` and certain files are being parsed to determine the list of `#include`d files so we can build the includePath for you.  This requires us to tag parse files in an order different than normal which is why the Parsing Workspace operation is paused.  If you have a bunch of 213MB header files that are taking a long time to parse, or are stuck, it will stall the whole system until those files are processed.  As a workaround, if you can use a more specific `includePath` that doesn't make use of the `**` feature, IntelliSense will no longer be blocked on this parsing.  (You can potentially copy paths from an existing log like you shared in your new issue)

**Comment by Colengms** [comment]
Closing this issue, based on the original SIGKILL repro having been addressed.
